File: 24.07.2024/queue.c
#include <stdio.h>
#define MAX 1000
int queue[MAX];
int front = -1, rear = -1;
void enqueue(int x) {
    if (rear >= MAX - 1) {
        printf("Queue Overflow\n");
        return;
    }
    if (front == -1) front = 0;
    queue[++rear] = x;
}
int dequeue() {
    if (front == -1 || front > rear) {
        printf("Queue Underflow\n");
        return -1;
    }
    return queue[front++];
}
int isEmpty() {
    return front == -1 || front > rear;
}
int main() {
    enqueue(10);
    enqueue(20);
    enqueue(30);
    printf("%d dequeued from queue\n", dequeue());
    printf("Front element is %d\n", queue[front]);
    return 0;
}

File: 24.07.2024/Queue.py
class Queue:
    def __init__(self):
        self.queue = []
    def enqueue(self, x):
        self.queue.append(x)
    def dequeue(self):
        if not self.is_empty():
            return self.queue.pop(0)
        return "Queue Underflow"
    def is_empty(self):
        return len(self.queue) == 0
q = Queue()
q.enqueue(10)
q.enqueue(20)
q.enqueue(30)
print(q.dequeue(), "dequeued from queue")
print("Front element is", q.queue[0] if not q.is_empty() else "Queue is empty")
